dts/bindings/regulator/nxp,pca9422.yaml

```diff
@@ -0,0 +1,165 @@
+# SPDX -License-Identifier: Apache-2.0
+
+description: |
+  NXP PCA9422 PMIC
+
+  The PMIC has one buck-boost converter, three buck converters, and four LDOs.
+  All need to be defined as children nodes, strictly following the BUCK-BOOST,
+  BUCK1, BUCK2, BUCK3, LDO1, LDO2, LDO3, and LDO4 node names.
+  For example:
+
+  pmic@61 {
+    reg = <0x61>;
+    ...
+    BUCK1 {
+      /* all properties for BUCK1 */
+    };
+    BUCK2 {
+      /* all properties for BUCK2 */
+    };
+    BUCK3 {
+      /* all properties for BUCK3 */
+    };
+    BUCKBOOST {
+      /* all properties for BUCKBOOST */
+    };
+    LDO1 {
+      /* all properties for LDO1 */
+    };
+    LDO2 {
+      /* all properties for LDO2 */
+    };
+    LDO3 {
+      /* all properties for LDO3 */
+    };
+    LDO4 {
+      /* all properties for LDO4 */
+    };
+  };
+
+
+compatible: "nxp,pca9422"
+
+include: base.yaml
+
+properties:
+  nxp,vin-ilim-microamp:
+    type: int
+    default: 470000
+    enum:
+      - 45000
+      - 70000
+      - 95000
+      - 120000
+      - 145000
+      - 170000
+      - 195000
+      - 220000
+      - 245000
+      - 270000
+      - 295000
+      - 320000
+      - 345000
+      - 370000
+      - 395000
+      - 420000
+      - 445000
+      - 470000
+      - 495000
+      - 520000
+      - 545000
+      - 570000
+      - 595000
+      - 620000
+      - 645000
+      - 670000
+      - 695000
+      - 795000
+      - 895000
+      - 995000
+      - 1095000
+      - 1195000
+    description: |
+      VIN input current limit, in microamperes. Value reflects typical value,
+      with accuraccy at +/-10%.
+
+      Defaults to 470mA, the IC default value.
+
+  nxp,vsys-uvlo-sel-millivolt:
+    type: int
+    default: 2700
+    enum:
+      - 2600
+      - 2700
+      - 2800
+    description: |
+      VSYS UVLO (under voltage lock out) threshold, in millivolts. Defaults to
+      2700mV to match the IC default value.
+
+child-binding:
+  include:
+    - name: regulator.yaml
+      property-allowlist:
+        - regulator-always-on
+        - regulator-boot-on
+        - regulator-min-microvolt
+        - regulator-max-microvolt
+
+  description: Voltage output of PMIC controller regulator
+  properties:
+    enable-inverted:
+      type: boolean
+      description: |
+        If the enable bit should be zero to turn the regulator on, add this
+        property.
+
+    nxp,mode0-microvolt:
```

Comment:
I don't think we should use the `mode0-mode3` naming scheme for this PMIC. The device has many additional modes, especially including the DVS states. Perhaps we label modes like this?:
- `dvs-0` (active DVS state, STBY_MODE1=L, SLEEP_MODE0=L)
- `sleep` (STBY_MODE1=L, SLEEP_MODE0=H)
- `standby` (STBY_MODE1=H, SLEEP_MODE0=L)
- `dpstandby` (STBY_MODE1=H, SLEEP_MODE0=H)

This will make defining additional DVS state mode names easier if support for this is added.

We also should consider how to handle DVS states, as these are only defined for the BUCK regulators (as far as I know). Perhaps we need a different binding for the buck versus LDO regulators?

Reply:
I was not attempting to implement the DVS modes of the PMIC at all. I was just trying to bring up the PCA9422 to parity with the PCA9420 driver and how the RT595 EVK uses it, which is only the two mode pins.

There has been a request, which is being discussed by the NXP PMIC team, to change the PCA9422 in the next silicon revision to have the modes be independent like they were in the PCA9420, in which case the more generic mode names would be more applicable.

Reply:
> I was not attempting to implement the DVS modes of the PMIC at all. I was just trying to bring up the PCA9422 to parity with the PCA9420 driver and how the RT595 EVK uses it, which is only the two mode pins.

Sure- that's totally fine. Just want to make sure this driver can be expanded in the future without changing devicetree binding names

> There has been a request, which is being discussed by the NXP PMIC team, to change the PCA9422 in the next silicon revision to have the modes be independent like they were in the PCA9420, in which case the more generic mode names would be more applicable.


IE in the next revision of the PCA9422 itself? My primary concern here is making sure the mode names match well with the datasheet itself, so that users don't have to dig into the driver code to understand what each mode name means

Reply:
If the next silicon revision doesn't make the modes more independent like they were in PCA9420, then I agree that it's better to make the mode names match the datasheet. My initial implementation was written in the hopes that they will make that change in the next silicon, so I wouldn't have to change the driver.

I'm not sure what to do at this point. Hold the PR until the silicon design team makes a decision? Or should I go ahead and make the driver match the current datasheet, knowing that big changes might happen in next revision? Guess that's the problem of writing drivers for not yet finalized silicon.

Reply:
> I'm not sure what to do at this point. Hold the PR until the silicon design team makes a decision? Or should I go ahead and make the driver match the current datasheet, knowing that big changes might happen in next revision? Guess that's the problem of writing drivers for not yet finalized silicon.

Do you know if the part number will change in the next silicon revision? I would say we should write the driver for the current revision of the silicon. If the next revision makes significant changes than we can update the driver to match. You have more visibility than I do into the possibility of this change though- if it is *very* likely, than perhaps keeping the mode names as is makes more sense